### PR TITLE
Rich history drawer: Translation added

### LINKED ISSUE #73981: i18n: Mark up Query History for translations
### Requirements ###

Add support for i18n in Query History components. All text presented in the UI should have i18n keys.

#### Details

* No translations need to be provided, just i18n keys
* Follow guidelines described in https://github.com/grafana/grafana/blob/main/contribute/internationalization.md
* Add `<Trans i18nKey ...` to all components in `explore/RichHistory` 
* Prefix `i18nKey` with `explore.history.`
* Don't change translation files. After adding new `<Trans i18nKey ...` run `yarn i18n:extract`

<details><summary>screenshots</summary>

Screenshot 2023-08-29 at 10 11 58

Screenshot 2023-08-29 at 10 12 04

</details>

#### Notes

* Change the language in user profile
* Translations are saved in public/locales/

🚧 Outstanding questions

...

### PR DESCRIPTION
**Pull request by siddhikhapare**
**What is this feature?**

Mark up Query History for translations.

**Why do we need this feature?**

To use the translation for query history components and achieve internationalization and localization.

**Who is this feature for?**

Users
**Which issue(s) does this PR fix?**:

Fixes #73981 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by siddhikhapare** [comment]
@ifrost Thank you for your guidance. I've updated the PR to address the necessary changes.

File: public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx [hunk 1]
```diff
@@ -241,11 +269,25 @@ export function RichHistoryQueriesTab(props: RichHistoryQueriesTabProps) {
...
           <div>
-            Showing {queries.length} of {totalQueries} <Button onClick={loadMoreRichHistory}>Load more</Button>
+            <Trans
+              i18nKey="explore.rich-history-queries-tab.showing-queries"
+              defaults="Showing {{ shown }} of {{ total }} <0>Load more</0>"
+              values={{ shown: queries.length, total: totalQueries }}
+              components={[
+                <Button onClick={loadMoreRichHistory} key="loadMoreButton">
+                  Load more
+                </Button>,
```
Comment (L-M-K-B) [inline public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:279]: Did you exclude this button from translations on purpose?

File: public/app/features/explore/RichHistory/RichHistorySettingsTab.tsx [hunk 2]
```diff
@@ -81,20 +91,32 @@ export function RichHistorySettingsTab(props: RichHistorySettingsProps) {
     <div className={styles.container}>
       {supportedFeatures().changeRetention ? (
         <Field
-          label="History time span"
-          description={`Select the period of time for which Grafana will save your query history. Up to ${MAX_HISTORY_ITEMS} entries will be stored.`}
+          label={t('explore.rich-history-settings-tab.history-time-span', 'History time span')}
+          description={t(
+            'explore.rich-history-settings-tab.history-time-span-description',
+            `Select the period of time for which Grafana will save your query history. Up to ${MAX_HISTORY_ITEMS} entries will be stored.`
+          )}
```
Comment (L-M-K-B) [inline public/app/features/explore/RichHistory/RichHistorySettingsTab.tsx:98]:
```suggestion
            'Select the period of time for which Grafana will save your query history. Up to {{MAX_HISTORY_ITEMS}} entries will be stored.', {MAX_HISTORY_ITEMS}
          )}
```

**Review by L-M-K-B — CHANGES_REQUESTED** [review]
Hi @siddhikhapare, Thanks a lot for contributing to Grafana. 
Your changes so far look great and I added  some comments below that I ask you to review.

File: public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx (hunk 1 above)
Comment (siddhikhapare) [reply public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:279]: @L-M-K-B No. I've used the approach mentioned in this GitHub link https://github.com/grafana/grafana/blob/327ae398e6438de7d8b76e55af6ae10d82749104/contribute/internationalization.md#react-components-and-html-tags for translation. I have used default translation template with placeholders for dynamic values ({{ shown }} and {{ total }}), HTML tags (<0>Load more</0>), and components.

**Comment by siddhikhapare** [comment]
@L-M-K-B I have updated changes. Please go through it once. Thank you.

File: public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx (hunk 1 above)
Comment (L-M-K-B) [reply public/app/features/explore/RichHistory/RichHistoryQueriesTab.tsx:279]: Ah, I see. Didn't notice the Trans component wrapping the button.

**Review by L-M-K-B — APPROVED** [review]

**Comment by L-M-K-B** [comment]
@siddhikhapare There are some failing tests in our build pipeline and I assume this is related to the node.js update we did yesterday. Can you please get the latest version of main from the Grafana repo and rebase your feature branch?

**Comment by siddhikhapare** [comment]
> @siddhikhapare There are some failing tests in our build pipeline and I assume this is related to the node.js update we did yesterday. Can you please get the latest version of main from the Grafana repo and rebase your feature branch?

@L-M-K-B I have updated changes. Thank you.

**Pull request merged**